### ISSUE TITLE
Ability to customize fallback locale

### ISSUE THREAD

**Issue by clarkwinkelmann** [flarum/core#1962]
This follows the discussion from #1774 in which @franzliedke mentions some missing feature regarding the fallback language.

The fallback locale is currently hard-coded here https://github.com/flarum/core/blob/f31f02d4cc0739c4c778a77f786d769634c32e10/src/Locale/LocaleServiceProvider.php#L62 as English.

It's already possible to customize it via `extend.php` by adding a `resolving` function to the container (I think). But it's not ideal.

We need to make it customizable in a user-friendly manner.

Use case: as Flarum becomes more and more international, we might start seeing extensions developed without any English language files. If, for example, many extensions were developed in Chinese only, Chinese forums that enabled multiple languages might want to set their default locale to Chinese so that the text from those extensions default to Chinese.

It seems to be possible to define multiple default locales, so we might want to allow setting multiple ones, maybe always including English as the last fallback.

Implementation ideas: we could start by making the default locale a customizable setting inside `config.php`. It could also be added to the admin panel.

If we can indeed set multiple fallback languages, we could set the default language chosen by the user as the first fallback language, followed by English. However this could have unintended side effects if users expect the English text to show up for untranslated texts. For example, a forum in French (default) and German might want English used for all extensions not translated in German instead of French.

So in my opinion English should always be the default fallback unless another locale has been explicitly set via a dedicated setting.

**Comment by franzliedke** [comment]
I agree with your suggestions. I hope the following sums it up correctly:
- keep English as the last-resort fallback
- allow optional configuration of an additional fallback language (user language -> fallback -> English) globally for the entire forum

I would suggest making this configurable in the admin panel, based on what language packs are installed.
